src/libecalc/common/variables.py

```diff
@@ -65,3 +66,45 @@ def get_subset_for_timestep(self, current_timestep: datetime) -> VariablesMap:
 
     def zeros(self) -> List[float]:
         return [0.0] * len(self.time_vector)
+
+    def get_variables(self):
+        return self.variables
+
+    def get_time_vector(self):
+        return self.time_vector
+
+    def get_period(self):
+        return self.period
+
+    def get_variables_map(self):
+        return self
+
+
+T = TypeVar(
+    "T",
+    bound=Union[List[datetime], Dict[str, List[Annotated[float, Field(allow_inf_nan=False)]]], Period, VariablesMap],
+    covariant=True,
+)
+
+
+class VariablesMapService(Protocol[T]):
+    @abc.abstractmethod
+    def get_time_vector(self) -> [List[datetime]]: ...
+
+    @abc.abstractmethod
+    def get_variables(self) -> Dict[str, List[Annotated[float, Field(allow_inf_nan=False)]]]: ...
```

Comment:
```suggestion
    def get_variables(self) -> Dict[str, List[float]]: ...
```

I don't think the annotations does anything here?